Content/ContentTypeResolver/SingleCategorySelectionResolver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Content\ContentTypeResolver;
+
+use JMS\Serializer\SerializationContext;
+use Sulu\Bundle\CategoryBundle\Category\CategoryManagerInterface;
+use Sulu\Bundle\HeadlessBundle\Content\ContentView;
+use Sulu\Bundle\HeadlessBundle\Content\Serializer\CategorySerializerInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+
+class SingleCategorySelectionResolver implements ContentTypeResolverInterface
+{
+    public static function getContentType(): string
+    {
+        return 'single_category_selection';
+    }
+
+    /**
+     * @var CategoryManagerInterface
+     */
+    private $categoryManager;
+
+    /**
+     * @var CategorySerializerInterface
+     */
+    private $categorySerializer;
+
+    public function __construct(
+        CategoryManagerInterface $categoryManager,
+        CategorySerializerInterface $categorySerializer
+    ) {
+        $this->categoryManager = $categoryManager;
+        $this->categorySerializer = $categorySerializer;
+    }
+
+    public function resolve($data, PropertyInterface $property, string $locale, array $attributes = []): ContentView
+    {
+        if (null === $data) {
+            return new ContentView([]);
+        }
+
+        $category = $this->categoryManager->findById((int) $data);
+        $serializationContext = new SerializationContext();
+        $serializationContext->setGroups(['partialCategory']);
+
+        $content = $this->categorySerializer->serialize($category, $locale, $serializationContext);
+
+        return new ContentView($content, $data ?? []);
```

Comment:
Some above would say a single selection should not return an array when not set /cc @nnatter

Reply:
the `$view` parameter of the `ContentView` needs to be an array at the moment and `[]` is the default value. but we just used `[$data]` in other single-selection resolvers. so i would do that here too 🙂 

https://github.com/sulu/SuluHeadlessBundle/blob/fd6fd7f0a6488cbdcef5ad279a72bce2f732e344/Content/ContentTypeResolver/SingleContactSelectionResolver.php#L59

Reply:
@nnatter make sense.
Implemented it =)